### PR TITLE
Add missing flow events for Signin Unblock

### LINKED ISSUE #1505: Ensure all signin unblock flow events are reported
Ensure we have [flow events for](https://github.com/mozilla/fxa/pull/194/files#diff-c84b6b3195c5290ee85400577cc04f23R36):

* [ ] Attempt to login (`flow.signin.submit`)
* [ ] Had login requests blocked for security reasons? (`account.login.blocked`)
* [x] Had unblock codes sent? (`account.login.sentUnblockCode`)
* [x] Successfully verified an unblock code? (`account.login.confirmedUnblockCode`)
* [ ] Used an invalid unblock code? (`account.login.invalidUnblockCode`)

ref https://github.com/mozilla/fxa-auth-server/pull/1471#issuecomment-251552928
ref https://github.com/mozilla/fxa-content-server/pull/4154

### PR DESCRIPTION
**Pull request by seanmonstar**
Adds the following flow events:

- `flow.signin.submit`
- `account.login.blocked`
- `account.login.invalidUnblockCode`

The follow are already in the code base:

- `account.login.sentUnblockCode`
- `account.login.confirmedUnblockCode`

Closes #1505 

r? @shane-tomlinson or @philbooth

### CODE REVIEW

File: lib/routes/account.js [hunk 1]
```diff
@@ -476,7 +479,8 @@ module.exports = function (
                     )
                     .catch(
                       (err) => {
-                        if (err.errno === error.ERRNO.UNBLOCK_CODE_INVALID) {
+                        if (err.errno === error.ERRNO.INVALID_UNBLOCK_CODE) {
```
Comment (seanmonstar) [inline lib/routes/account.js]: This typo means that currently, invalid codes usage doesn't get tracked in the customs server :cry:. Adding the tests means we caught it now.

File: lib/routes/account.js [hunk 2]
```diff
@@ -410,6 +410,8 @@ module.exports = function (
           })
         }
 
+        request.emitMetricsEvent('flow.signin.submit')
```
Comment (philbooth) [inline lib/routes/account.js]: Unfortunately this clashes with an event name we're already emitting from the content server. See the full list [here](https://github.com/mozilla/fxa-auth-server/blob/phil/issue-1366/docs/metrics-events.md#flow-events).

I think it's best if we reserve all `flow.*` names for the content server, seeing as that is how things have panned out so far and the distinction may help us avoid clashes in the future. Having a think about what I think this event should be now...

Comment (philbooth) [inline lib/routes/account.js]: > I think it's best if we reserve all `flow.*` names for the content server, seeing as that is how things have panned out so far and the distinction may help us avoid clashes in the future.

...except for `flow.complete`, duh.

Comment (philbooth) [inline lib/routes/account.js]: Maybe we should just delete this event, for now at least? We have an `attempt` event being emitted by the content server immediately prior to the request being sent, which combined with any error events we add to this route in due course should cover most of the drop-offs.

**Comment by philbooth** [comment]
Comment inline about the clashing event name but, that aside, this LGTM. r+

File: lib/routes/account.js (hunk 2 above)
Comment (seanmonstar) [inline lib/routes/account.js]: Dope, I thought the discussion in that issue was that we needed to add this event to the route. I'll just nuke the commit.

**Comment by shane-tomlinson** [comment]
🎆 Thanks @seanmonstar!

**Comment by shane-tomlinson** [comment]
Do these events need to be added to a document somewhere? @philbooth? What's the stance on documenting flow events?

**Comment by philbooth** [comment]
> Do these events need to be added to a document somewhere?

Indeed they do and I really should have been able to recall that when I reviewed this, given the doc was only merged yesterday. It's `docs/metrics-events.md` in this repo, going forward we should try to remember to update it at the same time as we add any events.

**Pull request merged**